src/hotspot/os/linux/gc/z/zPhysicalMemoryBacking_linux.cpp

```diff
@@ -204,12 +207,13 @@ int ZPhysicalMemoryBacking::create_mem_fd(const char* name) const {
   snprintf(filename, sizeof(filename), "%s%s", name, ZLargePages::is_explicit() ? ".hugetlb" : "");
 
   // Create file
-  const int extra_flags = ZLargePages::is_explicit() ? MFD_HUGETLB : 0;
+  const int extra_flags = ZLargePages::is_explicit() ? (MFD_HUGETLB | MFD_HUGE_2MB) : 0;
```

Comment:
Potentially the use of the constant `MFD_HUGE_2MB` could be generalized a little and calculated from a required page size like we do in `Linux::commit_memory_special` via a helper function like `os::Linux::hugetlbfs_page_size_flag`; at least that extra flag just looks like it is actually generated the same way. Looking through `memfd.h` it is *exactly* the same as for the corresponding `HUGETLB_*` flags.
But since this is really ZGC specific code, up to you.

Reply:
I don't think we should be passing `MAP_*` flags to `memfd_create()` as there's no guarantee that `MAP_HUGE_2MB` and `MFD_HUGE_2MB` are the same values. It's more of a lucky/convenient implementation detail that they happen to be the same.

Reply:
The manpages explicitly mention that the calculation is the same:
>       MFD_HUGE_2MB, MFD_HUGE_1GB, ...
>            Used in conjunction with MFD_HUGETLB to select alternative
>            hugetlb page sizes (respectively, 2 MB, 1 GB, ...)  on
>            systems that support multiple hugetlb page sizes.
>            Definitions for known huge page sizes are included in the
>            header file <linux/memfd.h>.
>
>            For details on encoding huge page sizes not included in
>            the header file, see the discussion of the similarly named
>            constants in mmap(2).

I.e. this is specified that way.

Reply:
Ok. I think I'll stick with an explicit `MFD_HUGE_2MB` since that's always what we want to use.